web/app/scripts/localization/date-picker-directive.js

```diff
@@ -0,0 +1,94 @@
+(function() {
+    'use strict';
+
+    /* ngInject */
+    function DatePickerController(DateLocalization, $log, $scope, $element) {
+        var ctl = this;
+
+        ctl.$onInit = initialize;
+
+        function initialize() {
+            // functions
+            ctl.updateDate = updateDate;
+
+            // data
+            ctl.dateConfig = DateLocalization.currentDateFormats();
```

Comment:
I'm feeling like a good chunk of this logic would be more appropriate in the `link` function of the directive, since a lot of it is the directive talking to itself rather than communicating with the outside world.